Mainnet/STOContract/STOContract/STOContract.cs

```diff
@@ -0,0 +1,1361 @@
+﻿using Stratis.SmartContracts;
+using Stratis.SmartContracts.Standards;
+
+[Deploy]
+public class STOContract : SmartContract
+{
+    public ulong EndBlock
+    {
+        get => this.PersistentState.GetUInt64(nameof(EndBlock));
+        private set => this.PersistentState.SetUInt64(nameof(EndBlock), value);
+    }
+
+    public Address TokenAddress
+    {
+        get => this.PersistentState.GetAddress(nameof(TokenAddress));
+        private set => this.PersistentState.SetAddress(nameof(TokenAddress), value);
+    }
+
+    public Address KYCAddress
+    {
+        get => this.PersistentState.GetAddress(nameof(KYCAddress));
+        private set => this.PersistentState.SetAddress(nameof(KYCAddress), value);
+    }
+
+    public Address MapperAddress
+    {
+        get => this.PersistentState.GetAddress(nameof(MapperAddress));
+        private set => this.PersistentState.SetAddress(nameof(MapperAddress), value);
+    }
+
+    public ulong TokenBalance
+    {
+        get => this.PersistentState.GetUInt64(nameof(TokenBalance));
+        private set => this.PersistentState.SetUInt64(nameof(TokenBalance), value);
+    }
+
+    public bool IsNonFungibleToken
+    {
+        get => this.PersistentState.GetBool(nameof(IsNonFungibleToken));
+        private set => this.PersistentState.SetBool(nameof(IsNonFungibleToken), value);
+    }
+
+    public bool SaleOpen => EndBlock >= this.Block.Number && TokenBalance > 0;
+
+    public Address Owner
+    {
+        get => this.PersistentState.GetAddress(nameof(Owner));
+        private set => this.PersistentState.SetAddress(nameof(Owner), value);
+    }
+
+    public SalePeriod[] SalePeriods
+    {
+        get => PersistentState.GetArray<SalePeriod>(nameof(SalePeriods));
+        private set => PersistentState.SetArray(nameof(SalePeriods), value);
+    }
+
+    public STOContract(ISmartContractState smartContractState,
+                       Address owner,
+                       uint tokenType,
+                       ulong totalSupply,
+                       string name,
+                       string symbol,
+                       Address kycAddress,
+                       Address mapperAddress,
+                       byte[] salePeriods) : base(smartContractState)
+    {
```

Comment:
I'll put together a PR to Stratis.SmartContracts.Standards to add the `Decimal` property along with a getter method. There is an existing one to add it to the Standard Token contract here: https://github.com/stratisproject/CirrusSmartContracts/pull/17/files that I believe should be voted in and have a `Decimals` property added to this constructor.

Also, the `ulong` to `uint256` switch depending on the internal decisions on that approach for token value properties.

Reply:
Contract updated with new types and decimals property